README.md

```diff
@@ -124,3 +124,131 @@ This is achieved by using Julia's native package extension mechanism. You can
 find all the current package integrations in the `src/QuartoNotebookWorker/ext`
 folder. Typically this is done via adding function hooks within the `__init__`
 method of the extension that run at different points during notebook execution.
+
+### Package Extensions
+
+As discussed above `QuartoNotebookWorker` is implemented as a full Julia package
+rather than just a `Module` loaded into the worker processes. This allows for
+any package to extend the functionality provided by the worker via Julia's
+[package extension mechanism][package-extensions]. For example, given a package
+called `PackageName` you could create a new package extension in
+`PackageName/ext/PackageNameQuartoNotebookWorkerExt.jl` with contents
+
+[package-extensions]: https://pkgdocs.julialang.org/v1/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)
+
+```julia
+module PackageNameQuartoNotebookWorkerExt
+
+import PackageName
+import QuartoNotebookWorker
+
+# ... Extension code here ...
+
+end
+```
+
+and update the `Project.toml` file for the `PackageName` package to include the
+following extension configuration:
+
+```toml
+[weakdeps]
+QuartoNotebookWorker = "38328d9c-a911-4051-bc06-3f7f556ffeda"
+
+[extensions]
+PackageNameQuartoNotebookWorkerExt = "QuartoNotebookWorker"
+```
+
+With these additions whenever `PackageName` is loaded into a `.qmd` file that is
+being run with `engine: julia` the extension code in the
+`PackageNameQuartoNotebookWorkerExt` module will be loaded. Below are the
+available interfaces that are can be extended.
+
+#### `expand`
+
+The `expand` function is used to inform `QuartoNotebookWorker` that a specific
+Julia type should not be rendered and instead should be converted into a series
+of notebook cells that are themselves evaluated and rendered. This allows for
+notebooks to generate a dynamic number of cells based on runtime information
+computed within the notebook rather than just the static cells of the original
+notebook source. In some sense you can think of this as macros for notebooks:
+code that generates code.
```

Comment:
I think we should make clearer that you don't really generate code with this feature like macros do, that it's all value-based